apis/api-journeys-modern/src/schema/journey/simple/updateSimpleJourney.ts

```diff
@@ -142,6 +143,47 @@ async function getYouTubeVideoDuration(videoId: string): Promise<number> {
   return parseISO8601Duration(isoDuration)
 }
 
+function getVideoId(video: JourneySimpleVideo): string {
+  if (video.source === 'youTube') {
+    const videoId = extractYouTubeVideoId(video.src)
+    if (videoId == null) {
+      throw new Error('Invalid YouTube video URL')
+    }
+    return videoId
+  }
+
+  if (video.source === 'internal') {
+    return video.src
+  }
+
+  throw new Error(`Unsupported video source: ${video.source as string}`)
+}
+
+async function getVideoEndAt(
+  video: JourneySimpleVideo,
+  videoId: string
+): Promise<number | undefined> {
+  if (video.source === 'youTube' && video.endAt == null) {
+    // youtube source and endAt not specified -> use full duration as end
+    return await getYouTubeVideoDuration(videoId)
+  }
+  return video.endAt
+}
+
+function getVideoVariantLanguageId(video: JourneySimpleVideo): string | null {
+  let videoVariantLanguageId: string | null
+  if (video.source === 'internal') {
+    if (video.subtitleId !== undefined && video.subtitleId !== null) {
+      videoVariantLanguageId = video.subtitleId
+    } else {
+      videoVariantLanguageId = '529'
+    }
+  } else {
+    videoVariantLanguageId = null
+  }
+  return videoVariantLanguageId
+}
```

Comment:
```suggestion
function getVideoVariantLanguageId(video: JourneySimpleVideo): string | null {
 if(video.source === 'youTube') return null
    if (video.subtitleId !== undefined && video.subtitleId !== null) {
      return video.subtitleId
    } 
      return '529'
  } 
}
```

thoughts on something like this?

Reply:
Thank you for this suggestion.

Made some extra adjustment to the suggested version, main difference is with the `null` and `undefined` checks.
By using a `!=` loose check we can cover both checks

```typescript
function getVideoVariantLanguageId(video: JourneySimpleVideo): string | null {
  if(video.source === 'youTube') return null
  if (video.subtitleId != null) return video.subtitleId
  return '529'
 }
 ```